### ISSUE TITLE
Implement API resources

### ISSUE THREAD

**Issue by meziantou** [meziantou/Meziantou.GitLabClient#2]
# Award Emoji
- [ ] [Issues, merge requests, and snippets - List an awardable's award emoji](https://docs.gitlab.com/ee/api/award_emoji.html#list-an-awardable-s-award-emoji)
- [ ] [Issues, merge requests, and snippets - Get single award emoji](https://docs.gitlab.com/ee/api/award_emoji.html#get-single-award-emoji)
- [ ] [Issues, merge requests, and snippets - Award a new emoji](https://docs.gitlab.com/ee/api/award_emoji.html#award-a-new-emoji)
- [ ] [Issues, merge requests, and snippets - Delete an award emoji](https://docs.gitlab.com/ee/api/award_emoji.html#delete-an-award-emoji)
- [ ] [Award Emoji on Notes - List a note's award emoji](https://docs.gitlab.com/ee/api/award_emoji.html#list-a-note-s-award-emoji)
- [ ] [Award Emoji on Notes - Get single note's award emoji](https://docs.gitlab.com/ee/api/award_emoji.html#get-single-note-s-award-emoji)
- [ ] [Award Emoji on Notes - Award a new emoji on a note](https://docs.gitlab.com/ee/api/award_emoji.html#award-a-new-emoji-on-a-note)

# Branches
- [ ] [List repository branches](https://docs.gitlab.com/ee/api/branches.html#list-repository-branches)
- [ ] [Get single repository branch](https://docs.gitlab.com/ee/api/branches.html#get-single-repository-branch)
- [ ] [Protect repository branch](https://docs.gitlab.com/ee/api/branches.html#protect-repository-branch)
- [ ] [Unprotect repository branch](https://docs.gitlab.com/ee/api/branches.html#unprotect-repository-branch)
- [ ] [Create repository branch](https://docs.gitlab.com/ee/api/branches.html#create-repository-branch)
- [ ] [Delete repository branch](https://docs.gitlab.com/ee/api/branches.html#delete-repository-branch)
- [ ] [Delete merged branches](https://docs.gitlab.com/ee/api/branches.html#delete-merged-branches)

# Broadcast Messages
- [ ] [Get all broadcast messages](https://docs.gitlab.com/ee/api/broadcast_messages.html#get-all-broadcast-messages)
- [ ] [Get a specific broadcast message](https://docs.gitlab.com/ee/api/broadcast_messages.html#get-a-specific-broadcast-message)
- [ ] [Create a broadcast message](https://docs.gitlab.com/ee/api/broadcast_messages.html#create-a-broadcast-message)
- [ ] [Update a broadcast message](https://docs.gitlab.com/ee/api/broadcast_messages.html#update-a-broadcast-message)
- [ ] [Delete a broadcast message](https://docs.gitlab.com/ee/api/broadcast_messages.html#delete-a-broadcast-message)

# Project-level Variables
- [ ] [List project variables](https://docs.gitlab.com/ee/api/project_level_variables.html#list-project-variables)
- [ ] [Show variable details](https://docs.gitlab.com/ee/api/project_level_variables.html#show-variable-details)
- [ ] [Create variable](https://docs.gitlab.com/ee/api/project_level_variables.html#create-variable)
- [ ] [Update variable](https://docs.gitlab.com/ee/api/project_level_variables.html#update-variable)
- [ ] [Remove variable](https://docs.gitlab.com/ee/api/project_level_variables.html#remove-variable)

# Group-level Variables
- [ ] [List group variables](https://docs.gitlab.com/ee/api/group_level_variables.html#list-group-variables)
- [ ] [Show variable details](https://docs.gitlab.com/ee/api/group_level_variables.html#show-variable-details)
- [ ] [Create variable](https://docs.gitlab.com/ee/api/group_level_variables.html#create-variable)
- [ ] [Update variable](https://docs.gitlab.com/ee/api/group_level_variables.html#update-variable)
- [ ] [Remove variable](https://docs.gitlab.com/ee/api/group_level_variables.html#remove-variable)

# Code Snippets
- [ ] [List snippets](https://docs.gitlab.com/ee/api/snippets.html#list-snippets)
- [ ] [Single snippet](https://docs.gitlab.com/ee/api/snippets.html#single-snippet)
- [ ] [Create new snippet](https://docs.gitlab.com/ee/api/snippets.html#create-new-snippet)
- [ ] [Update snippet](https://docs.gitlab.com/ee/api/snippets.html#update-snippet)
- [ ] [Delete snippet](https://docs.gitlab.com/ee/api/snippets.html#delete-snippet)
- [ ] [Explore all public snippets](https://docs.gitlab.com/ee/api/snippets.html#explore-all-public-snippets)
- [ ] [Get user agent details](https://docs.gitlab.com/ee/api/snippets.html#get-user-agent-details)

# Commits
- [ ] [List repository commits](https://docs.gitlab.com/ee/api/commits.html#list-repository-commits)
- [ ] [Create a commit with multiple files and actions](https://docs.gitlab.com/ee/api/commits.html#create-a-commit-with-multiple-files-and-actions)
- [ ] [Get a single commit](https://docs.gitlab.com/ee/api/commits.html#get-a-single-commit)
- [ ] [Get references a commit is pushed to](https://docs.gitlab.com/ee/api/commits.html#get-references-a-commit-is-pushed-to)
- [ ] [Cherry pick a commit](https://docs.gitlab.com/ee/api/commits.html#cherry-pick-a-commit)
- [ ] [Get the diff of a commit](https://docs.gitlab.com/ee/api/commits.html#get-the-diff-of-a-commit)
- [ ] [Get the comments of a commit](https://docs.gitlab.com/ee/api/commits.html#get-the-comments-of-a-commit)
- [ ] [Post comment to commit](https://docs.gitlab.com/ee/api/commits.html#post-comment-to-commit)
- [ ] [Commit status - List the statuses of a commit](https://docs.gitlab.com/ee/api/commits.html#list-the-statuses-of-a-commit)
- [ ] [Commit status - Post the build status to a commit](https://docs.gitlab.com/ee/api/commits.html#post-the-build-status-to-a-commit)
- [ ] [List Merge Requests associated with a commit](https://docs.gitlab.com/ee/api/commits.html#list-merge-requests-associated-with-a-commit)

# Custom Attributes
- [ ] [List custom attributes](https://docs.gitlab.com/ee/api/custom_attributes.html#list-custom-attributes)
- [ ] [Single custom attribute](https://docs.gitlab.com/ee/api/custom_attributes.html#single-custom-attribute)
- [ ] [Set custom attribute](https://docs.gitlab.com/ee/api/custom_attributes.html#set-custom-attribute)
- [ ] [Delete custom attribute](https://docs.gitlab.com/ee/api/custom_attributes.html#delete-custom-attribute)

# Deployments
- [ ] [List project deployments](https://docs.gitlab.com/ee/api/deployments.html#list-project-deployments)
- [ ] [Get a specific deployment](https://docs.gitlab.com/ee/api/deployments.html#get-a-specific-deployment)

# Deploy Keys
- [ ] [List all deploy keys](https://docs.gitlab.com/ee/api/deploy_keys.html#list-all-deploy-keys)
- [ ] [List project deploy keys](https://docs.gitlab.com/ee/api/deploy_keys.html#list-project-deploy-keys)
- [ ] [Single deploy key](https://docs.gitlab.com/ee/api/deploy_keys.html#single-deploy-key)
- [ ] [Add deploy key](https://docs.gitlab.com/ee/api/deploy_keys.html#add-deploy-key)
- [ ] [Update deploy key](https://docs.gitlab.com/ee/api/deploy_keys.html#update-deploy-key)
- [ ] [Delete deploy key](https://docs.gitlab.com/ee/api/deploy_keys.html#delete-deploy-key)

# Environments
- [ ] [List environments](https://docs.gitlab.com/ee/api/environments.html#list-environments)
- [ ] [Create a new environment](https://docs.gitlab.com/ee/api/environments.html#create-a-new-environment)
- [ ] [Edit an existing environment](https://docs.gitlab.com/ee/api/environments.html#edit-an-existing-environment)
- [ ] [Delete an environment](https://docs.gitlab.com/ee/api/environments.html#delete-an-environment)
- [ ] [Stop an environment](https://docs.gitlab.com/ee/api/environments.html#stop-an-environment)

# Epics
- [ ] [List epics for a group](https://docs.gitlab.com/ee/api/epics.html#list-epics-for-a-group)
- [ ] [Single epic](https://docs.gitlab.com/ee/api/epics.html#single-epic)
- [ ] [New epic](https://docs.gitlab.com/ee/api/epics.html#new-epic)
- [ ] [Update epic](https://docs.gitlab.com/ee/api/epics.html#update-epic)
- [ ] [Delete epic](https://docs.gitlab.com/ee/api/epics.html#delete-epic)
- [ ] [Create a todo](https://docs.gitlab.com/ee/api/epics.html#create-a-todo)

# Epic Issues
- [ ] [List issues for an epic](https://docs.gitlab.com/ee/api/epic_issues.html#list-issues-for-an-epic)
- [ ] [Assign an issue to the epic](https://docs.gitlab.com/ee/api/epic_issues.html#assign-an-issue-to-the-epic)
- [ ] [Remove an issue from the epic](https://docs.gitlab.com/ee/api/epic_issues.html#remove-an-issue-from-the-epic)
- [ ] [Update epic - issue association](https://docs.gitlab.com/ee/api/epic_issues.html#update-epic-issue-association)

# Events
- [ ] [List currently authenticated user's events](https://docs.gitlab.com/ee/api/events.html#list-currently-authenticated-user-s-events)
- [ ] [List currently authenticated user's events - Get user contribution events](https://docs.gitlab.com/ee/api/events.html#get-user-contribution-events)
- [ ] [List a Project's visible events](https://docs.gitlab.com/ee/api/events.html#list-a-project-s-visible-events)

# Feature flags
- [ ] [List all features](https://docs.gitlab.com/ee/api/features.html#list-all-features)
- [ ] [Set or create a feature](https://docs.gitlab.com/ee/api/features.html#set-or-create-a-feature)
- [ ] [Delete a feature](https://docs.gitlab.com/ee/api/features.html#delete-a-feature)

# Geo Nodes
- [ ] [Retrieve configuration about all Geo nodes](https://docs.gitlab.com/ee/api/geo_nodes.html#retrieve-configuration-about-all-geo-nodes)
- [ ] [Retrieve configuration about a specific Geo node](https://docs.gitlab.com/ee/api/geo_nodes.html#retrieve-configuration-about-a-specific-geo-node)
- [ ] [Edit a Geo node](https://docs.gitlab.com/ee/api/geo_nodes.html#edit-a-geo-node)
- [ ] [Delete a Geo node](https://docs.gitlab.com/ee/api/geo_nodes.html#delete-a-geo-node)
- [ ] [Repair a Geo node](https://docs.gitlab.com/ee/api/geo_nodes.html#repair-a-geo-node)
- [ ] [Retrieve status about all Geo nodes](https://docs.gitlab.com/ee/api/geo_nodes.html#retrieve-status-about-all-geo-nodes)
- [ ] [Retrieve status about a specific Geo node](https://docs.gitlab.com/ee/api/geo_nodes.html#retrieve-status-about-a-specific-geo-node)
- [ ] [Retrieve project sync or verification failures that occurred on the current node](https://docs.gitlab.com/ee/api/geo_nodes.html#retrieve-project-sync-or-verification-failures-that-occurred-on-the-current-node)

# Gitignores templates
- [ ] [List gitignore templates](https://docs.gitlab.com/ee/api/templates/gitignores.html#list-gitignore-templates)
- [ ] [Single gitignore template](https://docs.gitlab.com/ee/api/templates/gitignores.html#single-gitignore-template)

# GitLab CI Config templates
- [ ] [List GitLab CI YML templates](https://docs.gitlab.com/ee/api/templates/gitlab_ci_ymls.html#list-gitlab-ci-yml-templates)
- [ ] [Single GitLab CI YML template](https://docs.gitlab.com/ee/api/templates/gitlab_ci_ymls.html#single-gitlab-ci-yml-template)

# Groups
- [ ] [List groups](https://docs.gitlab.com/ee/api/groups.html#list-groups)
- [ ] [List a group's subgroups](https://docs.gitlab.com/ee/api/groups.html#list-a-group-s-subgroups)
- [ ] [List a group's projects](https://docs.gitlab.com/ee/api/groups.html#list-a-group-s-projects)
- [ ] [Details of a group](https://docs.gitlab.com/ee/api/groups.html#details-of-a-group)
- [ ] [New group](https://docs.gitlab.com/ee/api/groups.html#new-group)
- [ ] [Update group](https://docs.gitlab.com/ee/api/groups.html#update-group)
- [ ] [Remove group](https://docs.gitlab.com/ee/api/groups.html#remove-group)
- [ ] [Search for group](https://docs.gitlab.com/ee/api/groups.html#search-for-group)
- [ ] [Sync group with LDAP](https://docs.gitlab.com/ee/api/groups.html#sync-group-with-ldap)
- [ ] [Group members - Add LDAP group link](https://docs.gitlab.com/ee/api/groups.html#add-ldap-group-link)
- [ ] [Group members - Delete LDAP group link](https://docs.gitlab.com/ee/api/groups.html#delete-ldap-group-link)

# Group Access Requests
- [ ] [List access requests for a group or project](https://docs.gitlab.com/ee/api/access_requests.html#list-access-requests-for-a-group-or-project)
- [ ] [Request access to a group or project](https://docs.gitlab.com/ee/api/access_requests.html#request-access-to-a-group-or-project)
- [ ] [Approve an access request](https://docs.gitlab.com/ee/api/access_requests.html#approve-an-access-request)
- [ ] [Deny an access request](https://docs.gitlab.com/ee/api/access_requests.html#deny-an-access-request)

# Group Badges
- [ ] [List all badges of a group](https://docs.gitlab.com/ee/api/group_badges.html#list-all-badges-of-a-group)
- [ ] [Get a badge of a group](https://docs.gitlab.com/ee/api/group_badges.html#get-a-badge-of-a-group)
- [ ] [Add a badge to a group](https://docs.gitlab.com/ee/api/group_badges.html#add-a-badge-to-a-group)
- [ ] [Edit a badge of a group](https://docs.gitlab.com/ee/api/group_badges.html#edit-a-badge-of-a-group)
- [ ] [Remove a badge from a group](https://docs.gitlab.com/ee/api/group_badges.html#remove-a-badge-from-a-group)
- [ ] [Preview a badge from a group](https://docs.gitlab.com/ee/api/group_badges.html#preview-a-badge-from-a-group)

# Group Members
- [ ] [List all members of a group or project](https://docs.gitlab.com/ee/api/members.html#list-all-members-of-a-group-or-project)
- [ ] [List all members of a group or project including inherited members](https://docs.gitlab.com/ee/api/members.html#list-all-members-of-a-group-or-project-including-inherited-members)
- [ ] [Get a member of a group or project](https://docs.gitlab.com/ee/api/members.html#get-a-member-of-a-group-or-project)
- [ ] [Add a member to a group or project](https://docs.gitlab.com/ee/api/members.html#add-a-member-to-a-group-or-project)
- [ ] [Edit a member of a group or project](https://docs.gitlab.com/ee/api/members.html#edit-a-member-of-a-group-or-project)
- [ ] [Remove a member from a group or project](https://docs.gitlab.com/ee/api/members.html#remove-a-member-from-a-group-or-project)

# Issues
- [ ] [List issues](https://docs.gitlab.com/ee/api/issues.html#list-issues)
- [ ] [List group issues](https://docs.gitlab.com/ee/api/issues.html#list-group-issues)
- [ ] [List project issues](https://docs.gitlab.com/ee/api/issues.html#list-project-issues)
- [ ] [Single issue](https://docs.gitlab.com/ee/api/issues.html#single-issue)
- [x] [New issue](https://docs.gitlab.com/ee/api/issues.html#new-issue)
- [ ] [Edit issue](https://docs.gitlab.com/ee/api/issues.html#edit-issue)
- [ ] [Delete an issue](https://docs.gitlab.com/ee/api/issues.html#delete-an-issue)
- [ ] [Move an issue](https://docs.gitlab.com/ee/api/issues.html#move-an-issue)
- [ ] [Subscribe to an issue](https://docs.gitlab.com/ee/api/issues.html#subscribe-to-an-issue)
- [ ] [Unsubscribe from an issue](https://docs.gitlab.com/ee/api/issues.html#unsubscribe-from-an-issue)
- [ ] [Create a todo](https://docs.gitlab.com/ee/api/issues.html#create-a-todo)
- [ ] [Set a time estimate for an issue](https://docs.gitlab.com/ee/api/issues.html#set-a-time-estimate-for-an-issue)
- [ ] [Reset the time estimate for an issue](https://docs.gitlab.com/ee/api/issues.html#reset-the-time-estimate-for-an-issue)
- [ ] [Add spent time for an issue](https://docs.gitlab.com/ee/api/issues.html#add-spent-time-for-an-issue)
- [ ] [Reset spent time for an issue](https://docs.gitlab.com/ee/api/issues.html#reset-spent-time-for-an-issue)
- [ ] [Get time tracking stats](https://docs.gitlab.com/ee/api/issues.html#get-time-tracking-stats)
- [ ] [List merge requests that will close issue on merge](https://docs.gitlab.com/ee/api/issues.html#list-merge-requests-that-will-close-issue-on-merge)
- [ ] [Participants on issues](https://docs.gitlab.com/ee/api/issues.html#participants-on-issues)
- [ ] [Get user agent details](https://docs.gitlab.com/ee/api/issues.html#get-user-agent-details)

# Issue Boards
- [ ] [Project Board](https://docs.gitlab.com/ee/api/boards.html#project-board)
- [ ] [Single board](https://docs.gitlab.com/ee/api/boards.html#single-board)
- [ ] [Create a board](https://docs.gitlab.com/ee/api/boards.html#create-a-board)
- [ ] [Update a board](https://docs.gitlab.com/ee/api/boards.html#update-a-board)
- [ ] [Delete a board](https://docs.gitlab.com/ee/api/boards.html#delete-a-board)
- [ ] [List board lists](https://docs.gitlab.com/ee/api/boards.html#list-board-lists)
- [ ] [Single board list](https://docs.gitlab.com/ee/api/boards.html#single-board-list)
- [ ] [New board list](https://docs.gitlab.com/ee/api/boards.html#new-board-list)
- [ ] [Edit board list](https://docs.gitlab.com/ee/api/boards.html#edit-board-list)
- [ ] [Delete a board list](https://docs.gitlab.com/ee/api/boards.html#delete-a-board-list)

# Issue Links
- [ ] [List issue relations](https://docs.gitlab.com/ee/api/issue_links.html#list-issue-relations)
- [ ] [Create an issue link](https://docs.gitlab.com/ee/api/issue_links.html#create-an-issue-link)
- [ ] [Delete an issue link](https://docs.gitlab.com/ee/api/issue_links.html#delete-an-issue-link)

# Group Issue Boards
- [ ] [Group Board](https://docs.gitlab.com/ee/api/group_boards.html#group-board)
- [ ] [Single board](https://docs.gitlab.com/ee/api/group_boards.html#single-board)
- [ ] [Create a board](https://docs.gitlab.com/ee/api/group_boards.html#create-a-board)
- [ ] [Update a board](https://docs.gitlab.com/ee/api/group_boards.html#update-a-board)
- [ ] [Delete a board](https://docs.gitlab.com/ee/api/group_boards.html#delete-a-board)
- [ ] [List board lists](https://docs.gitlab.com/ee/api/group_boards.html#list-board-lists)
- [ ] [Single board list](https://docs.gitlab.com/ee/api/group_boards.html#single-board-list)
- [ ] [New board list](https://docs.gitlab.com/ee/api/group_boards.html#new-board-list)
- [ ] [Edit board list](https://docs.gitlab.com/ee/api/group_boards.html#edit-board-list)
- [ ] [Delete a board list](https://docs.gitlab.com/ee/api/group_boards.html#delete-a-board-list)

# Jobs
- [ ] [List project jobs](https://docs.gitlab.com/ee/api/jobs.html#list-project-jobs)
- [ ] [List pipeline jobs](https://docs.gitlab.com/ee/api/jobs.html#list-pipeline-jobs)
- [ ] [Get a single job](https://docs.gitlab.com/ee/api/jobs.html#get-a-single-job)
- [ ] [Get job artifacts](https://docs.gitlab.com/ee/api/jobs.html#get-job-artifacts)
- [ ] [Download the artifacts archive](https://docs.gitlab.com/ee/api/jobs.html#download-the-artifacts-archive)
- [ ] [Download a single artifact file](https://docs.gitlab.com/ee/api/jobs.html#download-a-single-artifact-file)
- [ ] [Get a trace file](https://docs.gitlab.com/ee/api/jobs.html#get-a-trace-file)
- [ ] [Cancel a job](https://docs.gitlab.com/ee/api/jobs.html#cancel-a-job)
- [ ] [Retry a job](https://docs.gitlab.com/ee/api/jobs.html#retry-a-job)
- [ ] [Erase a job](https://docs.gitlab.com/ee/api/jobs.html#erase-a-job)
- [ ] [Keep artifacts](https://docs.gitlab.com/ee/api/jobs.html#keep-artifacts)
- [ ] [Play a job](https://docs.gitlab.com/ee/api/jobs.html#play-a-job)

# Keys
- [ ] [Get SSH key with user by ID of an SSH key](https://docs.gitlab.com/ee/api/keys.html#get-ssh-key-with-user-by-id-of-an-ssh-key)

# Labels
- [ ] [List labels](https://docs.gitlab.com/ee/api/labels.html#list-labels)
- [ ] [Create a new label](https://docs.gitlab.com/ee/api/labels.html#create-a-new-label)
- [ ] [Delete a label](https://docs.gitlab.com/ee/api/labels.html#delete-a-label)
- [ ] [Edit an existing label](https://docs.gitlab.com/ee/api/labels.html#edit-an-existing-label)
- [ ] [Subscribe to a label](https://docs.gitlab.com/ee/api/labels.html#subscribe-to-a-label)
- [ ] [Unsubscribe from a label](https://docs.gitlab.com/ee/api/labels.html#unsubscribe-from-a-label)

# License
- [ ] [Retrieve information about the current license](https://docs.gitlab.com/ee/api/license.html#retrieve-information-about-the-current-license)
- [ ] [Add a new license](https://docs.gitlab.com/ee/api/license.html#add-a-new-license)

# Managed licenses
- [ ] [List managed licenses](https://docs.gitlab.com/ee/api/managed_licenses.html#list-managed-licenses)
- [ ] [Show an existing managed license](https://docs.gitlab.com/ee/api/managed_licenses.html#show-an-existing-managed-license)
- [ ] [Create a new managed license](https://docs.gitlab.com/ee/api/managed_licenses.html#create-a-new-managed-license)
- [ ] [Delete a managed license](https://docs.gitlab.com/ee/api/managed_licenses.html#delete-a-managed-license)

# Markdown
- [x] [Render an arbitrary Markdown document](https://docs.gitlab.com/ee/api/markdown.html#render-an-arbitrary-markdown-document)

# Merge Requests
- [x] [List merge requests](https://docs.gitlab.com/ee/api/merge_requests.html#list-merge-requests)
- [x] [List project merge requests](https://docs.gitlab.com/ee/api/merge_requests.html#list-project-merge-requests)
- [ ] [List group merge requests](https://docs.gitlab.com/ee/api/merge_requests.html#list-group-merge-requests)
- [x] [Get single MR](https://docs.gitlab.com/ee/api/merge_requests.html#get-single-mr)
- [ ] [Get single MR participants](https://docs.gitlab.com/ee/api/merge_requests.html#get-single-mr-participants)
- [ ] [Get single MR commits](https://docs.gitlab.com/ee/api/merge_requests.html#get-single-mr-commits)
- [ ] [Get single MR changes](https://docs.gitlab.com/ee/api/merge_requests.html#get-single-mr-changes)
- [ ] [List MR pipelines](https://docs.gitlab.com/ee/api/merge_requests.html#list-mr-pipelines)
- [x] [Create MR](https://docs.gitlab.com/ee/api/merge_requests.html#create-mr)
- [ ] [Update MR](https://docs.gitlab.com/ee/api/merge_requests.html#update-mr)
- [ ] [Delete a merge request](https://docs.gitlab.com/ee/api/merge_requests.html#delete-a-merge-request)
- [ ] [Accept MR](https://docs.gitlab.com/ee/api/merge_requests.html#accept-mr)
- [ ] [Cancel Merge When Pipeline Succeeds](https://docs.gitlab.com/ee/api/merge_requests.html#cancel-merge-when-pipeline-succeeds)
- [ ] [List issues that will close on merge](https://docs.gitlab.com/ee/api/merge_requests.html#list-issues-that-will-close-on-merge)
- [ ] [Subscribe to a merge request](https://docs.gitlab.com/ee/api/merge_requests.html#subscribe-to-a-merge-request)
- [ ] [Unsubscribe from a merge request](https://docs.gitlab.com/ee/api/merge_requests.html#unsubscribe-from-a-merge-request)
- [ ] [Create a todo](https://docs.gitlab.com/ee/api/merge_requests.html#create-a-todo)
- [ ] [Get MR diff versions](https://docs.gitlab.com/ee/api/merge_requests.html#get-mr-diff-versions)
- [ ] [Get a single MR diff version](https://docs.gitlab.com/ee/api/merge_requests.html#get-a-single-mr-diff-version)
- [ ] [Set a time estimate for a merge request](https://docs.gitlab.com/ee/api/merge_requests.html#set-a-time-estimate-for-a-merge-request)
- [ ] [Reset the time estimate for a merge request](https://docs.gitlab.com/ee/api/merge_requests.html#reset-the-time-estimate-for-a-merge-request)
- [ ] [Add spent time for a merge request](https://docs.gitlab.com/ee/api/merge_requests.html#add-spent-time-for-a-merge-request)
- [ ] [Reset spent time for a merge request](https://docs.gitlab.com/ee/api/merge_requests.html#reset-spent-time-for-a-merge-request)
- [ ] [Get time tracking stats](https://docs.gitlab.com/ee/api/merge_requests.html#get-time-tracking-stats)

# Merge Request Approvals
- [ ] [Project-level MR approvals - Get Configuration](https://docs.gitlab.com/ee/api/merge_request_approvals.html#get-configuration)
- [ ] [Project-level MR approvals - Change configuration](https://docs.gitlab.com/ee/api/merge_request_approvals.html#change-configuration)
- [ ] [Project-level MR approvals - Change allowed approvers](https://docs.gitlab.com/ee/api/merge_request_approvals.html#change-allowed-approvers)
- [ ] [Merge Request-level MR approvals - Change approval configuration](https://docs.gitlab.com/ee/api/merge_request_approvals.html#change-approval-configuration)
- [ ] [Merge Request-level MR approvals - Change allowed approvers for Merge Request](https://docs.gitlab.com/ee/api/merge_request_approvals.html#change-allowed-approvers-for-merge-request)
- [ ] [Approve Merge Request](https://docs.gitlab.com/ee/api/merge_request_approvals.html#approve-merge-request)
- [ ] [Unapprove Merge Request](https://docs.gitlab.com/ee/api/merge_request_approvals.html#unapprove-merge-request)

# Project milestones
- [ ] [List project milestones](https://docs.gitlab.com/ee/api/milestones.html#list-project-milestones)
- [ ] [Get single milestone](https://docs.gitlab.com/ee/api/milestones.html#get-single-milestone)
- [ ] [Create new milestone](https://docs.gitlab.com/ee/api/milestones.html#create-new-milestone)
- [ ] [Edit milestone](https://docs.gitlab.com/ee/api/milestones.html#edit-milestone)
- [ ] [Delete project milestone](https://docs.gitlab.com/ee/api/milestones.html#delete-project-milestone)
- [ ] [Get all issues assigned to a single milestone](https://docs.gitlab.com/ee/api/milestones.html#get-all-issues-assigned-to-a-single-milestone)
- [ ] [Get all merge requests assigned to a single milestone](https://docs.gitlab.com/ee/api/milestones.html#get-all-merge-requests-assigned-to-a-single-milestone)

# Group milestones
- [ ] [List group milestones](https://docs.gitlab.com/ee/api/group_milestones.html#list-group-milestones)
- [ ] [Get single milestone](https://docs.gitlab.com/ee/api/group_milestones.html#get-single-milestone)
- [ ] [Create new milestone](https://docs.gitlab.com/ee/api/group_milestones.html#create-new-milestone)
- [ ] [Edit milestone](https://docs.gitlab.com/ee/api/group_milestones.html#edit-milestone)
- [ ] [Delete group milestone](https://docs.gitlab.com/ee/api/group_milestones.html#delete-group-milestone)
- [ ] [Get all issues assigned to a single milestone](https://docs.gitlab.com/ee/api/group_milestones.html#get-all-issues-assigned-to-a-single-milestone)
- [ ] [Get all merge requests assigned to a single milestone](https://docs.gitlab.com/ee/api/group_milestones.html#get-all-merge-requests-assigned-to-a-single-milestone)

# Namespaces
- [ ] [List namespaces](https://docs.gitlab.com/ee/api/namespaces.html#list-namespaces)
- [ ] [Search for namespace](https://docs.gitlab.com/ee/api/namespaces.html#search-for-namespace)
- [ ] [Get namespace by ID](https://docs.gitlab.com/ee/api/namespaces.html#get-namespace-by-id)

# Notes
- [ ] [Issues - List project issue notes](https://docs.gitlab.com/ee/api/notes.html#list-project-issue-notes)
- [ ] [Issues - Get single issue note](https://docs.gitlab.com/ee/api/notes.html#get-single-issue-note)
- [ ] [Issues - Create new issue note](https://docs.gitlab.com/ee/api/notes.html#create-new-issue-note)
- [ ] [Issues - Modify existing issue note](https://docs.gitlab.com/ee/api/notes.html#modify-existing-issue-note)
- [ ] [Issues - Delete an issue note](https://docs.gitlab.com/ee/api/notes.html#delete-an-issue-note)
- [ ] [Snippets - List all snippet notes](https://docs.gitlab.com/ee/api/notes.html#list-all-snippet-notes)
- [ ] [Snippets - Get single snippet note](https://docs.gitlab.com/ee/api/notes.html#get-single-snippet-note)
- [ ] [Snippets - Create new snippet note](https://docs.gitlab.com/ee/api/notes.html#create-new-snippet-note)
- [ ] [Snippets - Modify existing snippet note](https://docs.gitlab.com/ee/api/notes.html#modify-existing-snippet-note)
- [ ] [Snippets - Delete a snippet note](https://docs.gitlab.com/ee/api/notes.html#delete-a-snippet-note)
- [ ] [Merge Requests - List all merge request notes](https://docs.gitlab.com/ee/api/notes.html#list-all-merge-request-notes)
- [ ] [Merge Requests - Get single merge request note](https://docs.gitlab.com/ee/api/notes.html#get-single-merge-request-note)
- [ ] [Merge Requests - Create new merge request note](https://docs.gitlab.com/ee/api/notes.html#create-new-merge-request-note)
- [ ] [Merge Requests - Modify existing merge request note](https://docs.gitlab.com/ee/api/notes.html#modify-existing-merge-request-note)
- [ ] [Merge Requests - Delete a merge request note](https://docs.gitlab.com/ee/api/notes.html#delete-a-merge-request-note)
- [ ] [Epics - List all epic notes](https://docs.gitlab.com/ee/api/notes.html#list-all-epic-notes)
- [ ] [Epics - Get single epic note](https://docs.gitlab.com/ee/api/notes.html#get-single-epic-note)
- [ ] [Epics - Create new epic note](https://docs.gitlab.com/ee/api/notes.html#create-new-epic-note)
- [ ] [Epics - Modify existing epic note](https://docs.gitlab.com/ee/api/notes.html#modify-existing-epic-note)
- [ ] [Epics - Delete an epic note](https://docs.gitlab.com/ee/api/notes.html#delete-an-epic-note)

# Discussions
- [ ] [Issues - List project issue discussions](https://docs.gitlab.com/ee/api/discussions.html#list-project-issue-discussions)
- [ ] [Issues - Get single issue discussion](https://docs.gitlab.com/ee/api/discussions.html#get-single-issue-discussion)
- [ ] [Issues - Create new issue discussion](https://docs.gitlab.com/ee/api/discussions.html#create-new-issue-discussion)
- [ ] [Issues - Add note to existing issue discussion](https://docs.gitlab.com/ee/api/discussions.html#add-note-to-existing-issue-discussion)
- [ ] [Issues - Modify existing issue discussion note](https://docs.gitlab.com/ee/api/discussions.html#modify-existing-issue-discussion-note)
- [ ] [Issues - Delete an issue discussion note](https://docs.gitlab.com/ee/api/discussions.html#delete-an-issue-discussion-note)
- [ ] [Snippets - List project snippet discussions](https://docs.gitlab.com/ee/api/discussions.html#list-project-snippet-discussions)
- [ ] [Snippets - Get single snippet discussion](https://docs.gitlab.com/ee/api/discussions.html#get-single-snippet-discussion)
- [ ] [Snippets - Create new snippet discussion](https://docs.gitlab.com/ee/api/discussions.html#create-new-snippet-discussion)
- [ ] [Snippets - Add note to existing snippet discussion](https://docs.gitlab.com/ee/api/discussions.html#add-note-to-existing-snippet-discussion)
- [ ] [Snippets - Modify existing snippet discussion note](https://docs.gitlab.com/ee/api/discussions.html#modify-existing-snippet-discussion-note)
- [ ] [Snippets - Delete an snippet discussion note](https://docs.gitlab.com/ee/api/discussions.html#delete-an-snippet-discussion-note)
- [ ] [Epics - List group epic discussions](https://docs.gitlab.com/ee/api/discussions.html#list-group-epic-discussions)
- [ ] [Epics - Get single epic discussion](https://docs.gitlab.com/ee/api/discussions.html#get-single-epic-discussion)
- [ ] [Epics - Create new epic discussion](https://docs.gitlab.com/ee/api/discussions.html#create-new-epic-discussion)
- [ ] [Epics - Add note to existing epic discussion](https://docs.gitlab.com/ee/api/discussions.html#add-note-to-existing-epic-discussion)
- [ ] [Epics - Modify existing epic discussion note](https://docs.gitlab.com/ee/api/discussions.html#modify-existing-epic-discussion-note)
- [ ] [Epics - Delete an epic discussion note](https://docs.gitlab.com/ee/api/discussions.html#delete-an-epic-discussion-note)
- [ ] [Merge requests - List project merge request discussions](https://docs.gitlab.com/ee/api/discussions.html#list-project-merge-request-discussions)
- [ ] [Merge requests - Get single merge request discussion](https://docs.gitlab.com/ee/api/discussions.html#get-single-merge-request-discussion)
- [ ] [Merge requests - Create new merge request discussion](https://docs.gitlab.com/ee/api/discussions.html#create-new-merge-request-discussion)
- [ ] [Merge requests - Resolve a merge request discussion](https://docs.gitlab.com/ee/api/discussions.html#resolve-a-merge-request-discussion)
- [ ] [Merge requests - Add note to existing merge request discussion](https://docs.gitlab.com/ee/api/discussions.html#add-note-to-existing-merge-request-discussion)
- [ ] [Merge requests - Modify an existing merge request discussion note](https://docs.gitlab.com/ee/api/discussions.html#modify-an-existing-merge-request-discussion-note)
- [ ] [Merge requests - Delete a merge request discussion note](https://docs.gitlab.com/ee/api/discussions.html#delete-a-merge-request-discussion-note)
- [ ] [Commits - List project commit discussions](https://docs.gitlab.com/ee/api/discussions.html#list-project-commit-discussions)
- [ ] [Commits - Get single commit discussion](https://docs.gitlab.com/ee/api/discussions.html#get-single-commit-discussion)
- [ ] [Commits - Create new commit discussion](https://docs.gitlab.com/ee/api/discussions.html#create-new-commit-discussion)
- [ ] [Commits - Add note to existing commit discussion](https://docs.gitlab.com/ee/api/discussions.html#add-note-to-existing-commit-discussion)
- [ ] [Commits - Modify an existing commit discussion note](https://docs.gitlab.com/ee/api/discussions.html#modify-an-existing-commit-discussion-note)
- [ ] [Commits - Delete a commit discussion note](https://docs.gitlab.com/ee/api/discussions.html#delete-a-commit-discussion-note)

# Resource Label Events
- [ ] [Issues - List project issue label events](https://docs.gitlab.com/ee/api/resource_label_events.html#list-project-issue-label-events)
- [ ] [Issues - Get single issue label event](https://docs.gitlab.com/ee/api/resource_label_events.html#get-single-issue-label-event)
- [ ] [Epics - List group epic label events](https://docs.gitlab.com/ee/api/resource_label_events.html#list-group-epic-label-events)
- [ ] [Epics - Get single epic label event](https://docs.gitlab.com/ee/api/resource_label_events.html#get-single-epic-label-event)
- [ ] [Merge requests - List project merge request label events](https://docs.gitlab.com/ee/api/resource_label_events.html#list-project-merge-request-label-events)
- [ ] [Merge requests - Get single merge request label event](https://docs.gitlab.com/ee/api/resource_label_events.html#get-single-merge-request-label-event)

# Notification settings
- [ ] [Global notification settings](https://docs.gitlab.com/ee/api/notification_settings.html#global-notification-settings)
- [ ] [Update global notification settings](https://docs.gitlab.com/ee/api/notification_settings.html#update-global-notification-settings)
- [ ] [Group / project level notification settings](https://docs.gitlab.com/ee/api/notification_settings.html#group-project-level-notification-settings)
- [ ] [Update group/project level notification settings](https://docs.gitlab.com/ee/api/notification_settings.html#update-group-project-level-notification-settings)

# Open source license templates
- [ ] [List license templates](https://docs.gitlab.com/ee/api/templates/licenses.html#list-license-templates)
- [ ] [Single license template](https://docs.gitlab.com/ee/api/templates/licenses.html#single-license-template)

# Pages Domains
- [ ] [List all pages domains](https://docs.gitlab.com/ee/api/pages_domains.html#list-all-pages-domains)
- [ ] [List pages domains](https://docs.gitlab.com/ee/api/pages_domains.html#list-pages-domains)
- [ ] [Single pages domain](https://docs.gitlab.com/ee/api/pages_domains.html#single-pages-domain)
- [ ] [Create new pages domain](https://docs.gitlab.com/ee/api/pages_domains.html#create-new-pages-domain)
- [ ] [Update pages domain](https://docs.gitlab.com/ee/api/pages_domains.html#update-pages-domain)
- [ ] [Delete pages domain](https://docs.gitlab.com/ee/api/pages_domains.html#delete-pages-domain)

# Pipelines
- [ ] [List project pipelines](https://docs.gitlab.com/ee/api/pipelines.html#list-project-pipelines)
- [ ] [Get a single pipeline](https://docs.gitlab.com/ee/api/pipelines.html#get-a-single-pipeline)
- [ ] [Create a new pipeline](https://docs.gitlab.com/ee/api/pipelines.html#create-a-new-pipeline)
- [ ] [Retry jobs in a pipeline](https://docs.gitlab.com/ee/api/pipelines.html#retry-jobs-in-a-pipeline)
- [ ] [Cancel a pipelines jobs](https://docs.gitlab.com/ee/api/pipelines.html#cancel-a-pipelines-jobs)

# Pipeline Triggers
- [ ] [List project triggers](https://docs.gitlab.com/ee/api/pipeline_triggers.html#list-project-triggers)
- [ ] [Get trigger details](https://docs.gitlab.com/ee/api/pipeline_triggers.html#get-trigger-details)
- [ ] [Create a project trigger](https://docs.gitlab.com/ee/api/pipeline_triggers.html#create-a-project-trigger)
- [ ] [Update a project trigger](https://docs.gitlab.com/ee/api/pipeline_triggers.html#update-a-project-trigger)
- [ ] [Take ownership of a project trigger](https://docs.gitlab.com/ee/api/pipeline_triggers.html#take-ownership-of-a-project-trigger)
- [ ] [Remove a project trigger](https://docs.gitlab.com/ee/api/pipeline_triggers.html#remove-a-project-trigger)

# Pipeline Schedules
- [ ] [Get all pipeline schedules](https://docs.gitlab.com/ee/api/pipeline_schedules.html#get-all-pipeline-schedules)
- [ ] [Get a single pipeline schedule](https://docs.gitlab.com/ee/api/pipeline_schedules.html#get-a-single-pipeline-schedule)
- [ ] [Create a new pipeline schedule](https://docs.gitlab.com/ee/api/pipeline_schedules.html#create-a-new-pipeline-schedule)
- [ ] [Edit a pipeline schedule](https://docs.gitlab.com/ee/api/pipeline_schedules.html#edit-a-pipeline-schedule)
- [ ] [Take ownership of a pipeline schedule](https://docs.gitlab.com/ee/api/pipeline_schedules.html#take-ownership-of-a-pipeline-schedule)
- [ ] [Delete a pipeline schedule](https://docs.gitlab.com/ee/api/pipeline_schedules.html#delete-a-pipeline-schedule)
- [ ] [Create a new pipeline schedule variable](https://docs.gitlab.com/ee/api/pipeline_schedules.html#create-a-new-pipeline-schedule-variable)
- [ ] [Edit a pipeline schedule variable](https://docs.gitlab.com/ee/api/pipeline_schedules.html#edit-a-pipeline-schedule-variable)
- [ ] [Delete a pipeline schedule variable](https://docs.gitlab.com/ee/api/pipeline_schedules.html#delete-a-pipeline-schedule-variable)

# Projects
- [x] [List all projects](https://docs.gitlab.com/ee/api/projects.html#list-all-projects)
- [x] [List user projects](https://docs.gitlab.com/ee/api/projects.html#list-user-projects)
- [x] [Get single project](https://docs.gitlab.com/ee/api/projects.html#get-single-project)
- [ ] [Get project users](https://docs.gitlab.com/ee/api/projects.html#get-project-users)
- [x] [Create project](https://docs.gitlab.com/ee/api/projects.html#create-project)
- [ ] [Create project for user](https://docs.gitlab.com/ee/api/projects.html#create-project-for-user)
- [ ] [Edit project](https://docs.gitlab.com/ee/api/projects.html#edit-project)
- [ ] [Fork project](https://docs.gitlab.com/ee/api/projects.html#fork-project)
- [ ] [List Forks of a project](https://docs.gitlab.com/ee/api/projects.html#list-forks-of-a-project)
- [ ] [Star a project](https://docs.gitlab.com/ee/api/projects.html#star-a-project)
- [ ] [Unstar a project](https://docs.gitlab.com/ee/api/projects.html#unstar-a-project)
- [ ] [Languages](https://docs.gitlab.com/ee/api/projects.html#languages)
- [ ] [Archive a project](https://docs.gitlab.com/ee/api/projects.html#archive-a-project)
- [ ] [Unarchive a project](https://docs.gitlab.com/ee/api/projects.html#unarchive-a-project)
- [ ] [Remove project](https://docs.gitlab.com/ee/api/projects.html#remove-project)
- [ ] [Upload a file](https://docs.gitlab.com/ee/api/projects.html#upload-a-file)
- [ ] [Share project with group](https://docs.gitlab.com/ee/api/projects.html#share-project-with-group)
- [ ] [Delete a shared project link within a group](https://docs.gitlab.com/ee/api/projects.html#delete-a-shared-project-link-within-a-group)
- [ ] [Hooks - List project hooks](https://docs.gitlab.com/ee/api/projects.html#list-project-hooks)
- [ ] [Hooks - Get project hook](https://docs.gitlab.com/ee/api/projects.html#get-project-hook)
- [ ] [Hooks - Add project hook](https://docs.gitlab.com/ee/api/projects.html#add-project-hook)
- [ ] [Hooks - Edit project hook](https://docs.gitlab.com/ee/api/projects.html#edit-project-hook)
- [ ] [Hooks - Delete project hook](https://docs.gitlab.com/ee/api/projects.html#delete-project-hook)
- [ ] [Fork relationship - Create a forked from/to relation between existing projects](https://docs.gitlab.com/ee/api/projects.html#create-a-forked-from-to-relation-between-existing-projects)
- [ ] [Fork relationship - Delete an existing forked from relationship](https://docs.gitlab.com/ee/api/projects.html#delete-an-existing-forked-from-relationship)
- [ ] [Search for projects by name](https://docs.gitlab.com/ee/api/projects.html#search-for-projects-by-name)
- [ ] [Start the Housekeeping task for a Project](https://docs.gitlab.com/ee/api/projects.html#start-the-housekeeping-task-for-a-project)
- [ ] [Push Rules - Get project push rules](https://docs.gitlab.com/ee/api/projects.html#get-project-push-rules)
- [ ] [Push Rules - Add project push rule](https://docs.gitlab.com/ee/api/projects.html#add-project-push-rule)
- [ ] [Push Rules - Edit project push rule](https://docs.gitlab.com/ee/api/projects.html#edit-project-push-rule)
- [ ] [Push Rules - Delete project push rule](https://docs.gitlab.com/ee/api/projects.html#delete-project-push-rule)
- [ ] [Transfer a project to a new namespace](https://docs.gitlab.com/ee/api/projects.html#transfer-a-project-to-a-new-namespace)
- [ ] [Start the pull mirroring process for a Project](https://docs.gitlab.com/ee/api/projects.html#start-the-pull-mirroring-process-for-a-project)
- [ ] [Download snapshot of a git repository](https://docs.gitlab.com/ee/api/projects.html#download-snapshot-of-a-git-repository)

# Project Access Requests
- [ ] [List access requests for a group or project](https://docs.gitlab.com/ee/api/access_requests.html#list-access-requests-for-a-group-or-project)
- [ ] [Request access to a group or project](https://docs.gitlab.com/ee/api/access_requests.html#request-access-to-a-group-or-project)
- [ ] [Approve an access request](https://docs.gitlab.com/ee/api/access_requests.html#approve-an-access-request)
- [ ] [Deny an access request](https://docs.gitlab.com/ee/api/access_requests.html#deny-an-access-request)

# Project Badges
- [ ] [List all badges of a project](https://docs.gitlab.com/ee/api/project_badges.html#list-all-badges-of-a-project)
- [ ] [Get a badge of a project](https://docs.gitlab.com/ee/api/project_badges.html#get-a-badge-of-a-project)
- [ ] [Add a badge to a project](https://docs.gitlab.com/ee/api/project_badges.html#add-a-badge-to-a-project)
- [ ] [Edit a badge of a project](https://docs.gitlab.com/ee/api/project_badges.html#edit-a-badge-of-a-project)
- [ ] [Remove a badge from a project](https://docs.gitlab.com/ee/api/project_badges.html#remove-a-badge-from-a-project)
- [ ] [Preview a badge from a project](https://docs.gitlab.com/ee/api/project_badges.html#preview-a-badge-from-a-project)

# Project import/export
- [ ] [Schedule an export](https://docs.gitlab.com/ee/api/project_import_export.html#schedule-an-export)
- [ ] [Export status](https://docs.gitlab.com/ee/api/project_import_export.html#export-status)
- [ ] [Export download](https://docs.gitlab.com/ee/api/project_import_export.html#export-download)
- [ ] [Import a file](https://docs.gitlab.com/ee/api/project_import_export.html#import-a-file)
- [ ] [Import status](https://docs.gitlab.com/ee/api/project_import_export.html#import-status)

# Project Members
- [ ] [List all members of a group or project](https://docs.gitlab.com/ee/api/members.html#list-all-members-of-a-group-or-project)
- [ ] [List all members of a group or project including inherited members](https://docs.gitlab.com/ee/api/members.html#list-all-members-of-a-group-or-project-including-inherited-members)
- [ ] [Get a member of a group or project](https://docs.gitlab.com/ee/api/members.html#get-a-member-of-a-group-or-project)
- [ ] [Add a member to a group or project](https://docs.gitlab.com/ee/api/members.html#add-a-member-to-a-group-or-project)
- [ ] [Edit a member of a group or project](https://docs.gitlab.com/ee/api/members.html#edit-a-member-of-a-group-or-project)
- [ ] [Remove a member from a group or project](https://docs.gitlab.com/ee/api/members.html#remove-a-member-from-a-group-or-project)

# Project Snippets
- [ ] [List snippets](https://docs.gitlab.com/ee/api/project_snippets.html#list-snippets)
- [ ] [Single snippet](https://docs.gitlab.com/ee/api/project_snippets.html#single-snippet)
- [ ] [Create new snippet](https://docs.gitlab.com/ee/api/project_snippets.html#create-new-snippet)
- [ ] [Update snippet](https://docs.gitlab.com/ee/api/project_snippets.html#update-snippet)
- [ ] [Delete snippet](https://docs.gitlab.com/ee/api/project_snippets.html#delete-snippet)
- [ ] [Snippet content](https://docs.gitlab.com/ee/api/project_snippets.html#snippet-content)
- [ ] [Get user agent details](https://docs.gitlab.com/ee/api/project_snippets.html#get-user-agent-details)

# Protected Branches
- [ ] [List protected branches](https://docs.gitlab.com/ee/api/protected_branches.html#list-protected-branches)
- [ ] [Get a single protected branch or wildcard protected branch](https://docs.gitlab.com/ee/api/protected_branches.html#get-a-single-protected-branch-or-wildcard-protected-branch)
- [ ] [Protect repository branches](https://docs.gitlab.com/ee/api/protected_branches.html#protect-repository-branches)
- [ ] [Unprotect repository branches](https://docs.gitlab.com/ee/api/protected_branches.html#unprotect-repository-branches)

# Protected Tags
- [ ] [List protected tags](https://docs.gitlab.com/ee/api/protected_tags.html#list-protected-tags)
- [ ] [Get a single protected tag or wildcard protected tag](https://docs.gitlab.com/ee/api/protected_tags.html#get-a-single-protected-tag-or-wildcard-protected-tag)
- [ ] [Protect repository tags](https://docs.gitlab.com/ee/api/protected_tags.html#protect-repository-tags)
- [ ] [Unprotect repository tags](https://docs.gitlab.com/ee/api/protected_tags.html#unprotect-repository-tags)

# Repositories
- [ ] [List repository tree](https://docs.gitlab.com/ee/api/repositories.html#list-repository-tree)
- [ ] [Get a blob from repository](https://docs.gitlab.com/ee/api/repositories.html#get-a-blob-from-repository)
- [ ] [Raw blob content](https://docs.gitlab.com/ee/api/repositories.html#raw-blob-content)
- [ ] [Get file archive](https://docs.gitlab.com/ee/api/repositories.html#get-file-archive)
- [ ] [Compare branches, tags or commits](https://docs.gitlab.com/ee/api/repositories.html#compare-branches-tags-or-commits)
- [ ] [Contributors](https://docs.gitlab.com/ee/api/repositories.html#contributors)
- [ ] [Merge Base](https://docs.gitlab.com/ee/api/repositories.html#merge-base)

# Repository Files
- [ ] [Get file from repository](https://docs.gitlab.com/ee/api/repository_files.html#get-file-from-repository)
- [ ] [Get raw file from repository](https://docs.gitlab.com/ee/api/repository_files.html#get-raw-file-from-repository)
- [x] [Create new file in repository](https://docs.gitlab.com/ee/api/repository_files.html#create-new-file-in-repository)
- [x] [Update existing file in repository](https://docs.gitlab.com/ee/api/repository_files.html#update-existing-file-in-repository)
- [ ] [Delete existing file in repository](https://docs.gitlab.com/ee/api/repository_files.html#delete-existing-file-in-repository)

# Runners
- [ ] [List owned runners](https://docs.gitlab.com/ee/api/runners.html#list-owned-runners)
- [ ] [List all runners](https://docs.gitlab.com/ee/api/runners.html#list-all-runners)
- [ ] [Get runner's details](https://docs.gitlab.com/ee/api/runners.html#get-runner-s-details)
- [ ] [Update runner's details](https://docs.gitlab.com/ee/api/runners.html#update-runner-s-details)
- [ ] [Remove a runner](https://docs.gitlab.com/ee/api/runners.html#remove-a-runner)
- [ ] [List runner's jobs](https://docs.gitlab.com/ee/api/runners.html#list-runner-s-jobs)
- [ ] [List project's runners](https://docs.gitlab.com/ee/api/runners.html#list-project-s-runners)
- [ ] [Enable a runner in project](https://docs.gitlab.com/ee/api/runners.html#enable-a-runner-in-project)
- [ ] [Disable a runner from project](https://docs.gitlab.com/ee/api/runners.html#disable-a-runner-from-project)
- [ ] [Register a new Runner](https://docs.gitlab.com/ee/api/runners.html#register-a-new-runner)
- [ ] [Delete a registered Runner](https://docs.gitlab.com/ee/api/runners.html#delete-a-registered-runner)
- [ ] [Verify authentication for a registered Runner](https://docs.gitlab.com/ee/api/runners.html#verify-authentication-for-a-registered-runner)

# Search
- [ ] [Global Search API](https://docs.gitlab.com/ee/api/search.html#global-search-api)
- [ ] [Group Search API](https://docs.gitlab.com/ee/api/search.html#group-search-api)
- [ ] [Project Search API](https://docs.gitlab.com/ee/api/search.html#project-search-api)

# Services
- [ ] [Asana - Create/Edit Asana service](https://docs.gitlab.com/ee/api/services.html#create-edit-asana-service)
- [ ] [Asana - Delete Asana service](https://docs.gitlab.com/ee/api/services.html#delete-asana-service)
- [ ] [Asana - Get Asana service settings](https://docs.gitlab.com/ee/api/services.html#get-asana-service-settings)
- [ ] [Assembla - Create/Edit Assembla service](https://docs.gitlab.com/ee/api/services.html#create-edit-assembla-service)
- [ ] [Assembla - Delete Assembla service](https://docs.gitlab.com/ee/api/services.html#delete-assembla-service)
- [ ] [Assembla - Get Assembla service settings](https://docs.gitlab.com/ee/api/services.html#get-assembla-service-settings)
- [ ] [Atlassian Bamboo CI - Create/Edit Atlassian Bamboo CI service](https://docs.gitlab.com/ee/api/services.html#create-edit-atlassian-bamboo-ci-service)
- [ ] [Atlassian Bamboo CI - Delete Atlassian Bamboo CI service](https://docs.gitlab.com/ee/api/services.html#delete-atlassian-bamboo-ci-service)
- [ ] [Atlassian Bamboo CI - Get Atlassian Bamboo CI service settings](https://docs.gitlab.com/ee/api/services.html#get-atlassian-bamboo-ci-service-settings)
- [ ] [Bugzilla - Create/Edit Buildkite service](https://docs.gitlab.com/ee/api/services.html#create-edit-buildkite-service)
- [ ] [Bugzilla - Delete Bugzilla Service](https://docs.gitlab.com/ee/api/services.html#delete-bugzilla-service)
- [ ] [Bugzilla - Get Bugzilla Service Settings](https://docs.gitlab.com/ee/api/services.html#get-bugzilla-service-settings)
- [ ] [Buildkite - Delete Buildkite service](https://docs.gitlab.com/ee/api/services.html#delete-buildkite-service)
- [ ] [Buildkite - Get Buildkite service settings](https://docs.gitlab.com/ee/api/services.html#get-buildkite-service-settings)
- [ ] [Campfire - Create/Edit Campfire service](https://docs.gitlab.com/ee/api/services.html#create-edit-campfire-service)
- [ ] [Campfire - Delete Campfire service](https://docs.gitlab.com/ee/api/services.html#delete-campfire-service)
- [ ] [Campfire - Get Campfire service settings](https://docs.gitlab.com/ee/api/services.html#get-campfire-service-settings)
- [ ] [Custom Issue Tracker - Create/Edit Custom Issue Tracker service](https://docs.gitlab.com/ee/api/services.html#create-edit-custom-issue-tracker-service)
- [ ] [Custom Issue Tracker - Delete Custom Issue Tracker service](https://docs.gitlab.com/ee/api/services.html#delete-custom-issue-tracker-service)
- [ ] [Custom Issue Tracker - Get Custom Issue Tracker service settings](https://docs.gitlab.com/ee/api/services.html#get-custom-issue-tracker-service-settings)
- [ ] [Drone CI - Create/Edit Drone CI service](https://docs.gitlab.com/ee/api/services.html#create-edit-drone-ci-service)
- [ ] [Drone CI - Delete Drone CI service](https://docs.gitlab.com/ee/api/services.html#delete-drone-ci-service)
- [ ] [Drone CI - Get Drone CI service settings](https://docs.gitlab.com/ee/api/services.html#get-drone-ci-service-settings)
- [ ] [Emails on push - Create/Edit Emails on push service](https://docs.gitlab.com/ee/api/services.html#create-edit-emails-on-push-service)
- [ ] [Emails on push - Delete Emails on push service](https://docs.gitlab.com/ee/api/services.html#delete-emails-on-push-service)
- [ ] [Emails on push - Get Emails on push service settings](https://docs.gitlab.com/ee/api/services.html#get-emails-on-push-service-settings)
- [ ] [External Wiki - Create/Edit External Wiki service](https://docs.gitlab.com/ee/api/services.html#create-edit-external-wiki-service)
- [ ] [External Wiki - Delete External Wiki service](https://docs.gitlab.com/ee/api/services.html#delete-external-wiki-service)
- [ ] [External Wiki - Get External Wiki service settings](https://docs.gitlab.com/ee/api/services.html#get-external-wiki-service-settings)
- [ ] [Flowdock - Create/Edit Flowdock service](https://docs.gitlab.com/ee/api/services.html#create-edit-flowdock-service)
- [ ] [Flowdock - Delete Flowdock service](https://docs.gitlab.com/ee/api/services.html#delete-flowdock-service)
- [ ] [Flowdock - Get Flowdock service settings](https://docs.gitlab.com/ee/api/services.html#get-flowdock-service-settings)
- [ ] [Hangouts Chat - Create/Edit Hangouts Chat service](https://docs.gitlab.com/ee/api/services.html#create-edit-hangouts-chat-service)
- [ ] [Hangouts Chat - Delete Hangouts Chat service](https://docs.gitlab.com/ee/api/services.html#delete-hangouts-chat-service)
- [ ] [Hangouts Chat - Get Hangouts Chat service settings](https://docs.gitlab.com/ee/api/services.html#get-hangouts-chat-service-settings)
- [ ] [HipChat - Create/Edit HipChat service](https://docs.gitlab.com/ee/api/services.html#create-edit-hipchat-service)
- [ ] [HipChat - Delete HipChat service](https://docs.gitlab.com/ee/api/services.html#delete-hipchat-service)
- [ ] [HipChat - Get HipChat service settings](https://docs.gitlab.com/ee/api/services.html#get-hipchat-service-settings)
- [ ] [Irker (IRC gateway) - Create/Edit Irker (IRC gateway) service](https://docs.gitlab.com/ee/api/services.html#create-edit-irker-irc-gateway-service)
- [ ] [Irker (IRC gateway) - Delete Irker (IRC gateway) service](https://docs.gitlab.com/ee/api/services.html#delete-irker-irc-gateway-service)
- [ ] [Irker (IRC gateway) - Get Irker (IRC gateway) service settings](https://docs.gitlab.com/ee/api/services.html#get-irker-irc-gateway-service-settings)
- [ ] [JIRA - Get JIRA service settings](https://docs.gitlab.com/ee/api/services.html#get-jira-service-settings)
- [ ] [JIRA - Create/Edit JIRA service](https://docs.gitlab.com/ee/api/services.html#create-edit-jira-service)
- [ ] [JIRA - Delete JIRA service](https://docs.gitlab.com/ee/api/services.html#delete-jira-service)
- [ ] [Kubernetes - Create/Edit Kubernetes service](https://docs.gitlab.com/ee/api/services.html#create-edit-kubernetes-service)
- [ ] [Kubernetes - Delete Kubernetes service](https://docs.gitlab.com/ee/api/services.html#delete-kubernetes-service)
- [ ] [Kubernetes - Get Kubernetes service settings](https://docs.gitlab.com/ee/api/services.html#get-kubernetes-service-settings)
- [ ] [Slack slash commands - Get Slack slash command service settings](https://docs.gitlab.com/ee/api/services.html#get-slack-slash-command-service-settings)
- [ ] [Slack slash commands - Create/Edit Slack slash command service](https://docs.gitlab.com/ee/api/services.html#create-edit-slack-slash-command-service)
- [ ] [Slack slash commands - Delete Slack slash command service](https://docs.gitlab.com/ee/api/services.html#delete-slack-slash-command-service)
- [ ] [Mattermost slash commands - Get Mattermost slash command service settings](https://docs.gitlab.com/ee/api/services.html#get-mattermost-slash-command-service-settings)
- [ ] [Mattermost slash commands - Create/Edit Mattermost slash command service](https://docs.gitlab.com/ee/api/services.html#create-edit-mattermost-slash-command-service)
- [ ] [Mattermost slash commands - Delete Mattermost slash command service](https://docs.gitlab.com/ee/api/services.html#delete-mattermost-slash-command-service)
- [ ] [Packagist - Create/Edit Packagist service](https://docs.gitlab.com/ee/api/services.html#create-edit-packagist-service)
- [ ] [Packagist - Delete Packagist service](https://docs.gitlab.com/ee/api/services.html#delete-packagist-service)
- [ ] [Packagist - Get Packagist service settings](https://docs.gitlab.com/ee/api/services.html#get-packagist-service-settings)
- [ ] [Pipeline-Emails - Create/Edit Pipeline-Emails service](https://docs.gitlab.com/ee/api/services.html#create-edit-pipeline-emails-service)
- [ ] [Pipeline-Emails - Delete Pipeline-Emails service](https://docs.gitlab.com/ee/api/services.html#delete-pipeline-emails-service)
- [ ] [Pipeline-Emails - Get Pipeline-Emails service settings](https://docs.gitlab.com/ee/api/services.html#get-pipeline-emails-service-settings)
- [ ] [PivotalTracker - Create/Edit PivotalTracker service](https://docs.gitlab.com/ee/api/services.html#create-edit-pivotaltracker-service)
- [ ] [PivotalTracker - Delete PivotalTracker service](https://docs.gitlab.com/ee/api/services.html#delete-pivotaltracker-service)
- [ ] [PivotalTracker - Get PivotalTracker service settings](https://docs.gitlab.com/ee/api/services.html#get-pivotaltracker-service-settings)
- [ ] [Prometheus - Create/Edit Prometheus service](https://docs.gitlab.com/ee/api/services.html#create-edit-prometheus-service)
- [ ] [Prometheus - Delete Prometheus service](https://docs.gitlab.com/ee/api/services.html#delete-prometheus-service)
- [ ] [Prometheus - Get Prometheus service settings](https://docs.gitlab.com/ee/api/services.html#get-prometheus-service-settings)
- [ ] [Pushover - Create/Edit Pushover service](https://docs.gitlab.com/ee/api/services.html#create-edit-pushover-service)
- [ ] [Pushover - Delete Pushover service](https://docs.gitlab.com/ee/api/services.html#delete-pushover-service)
- [ ] [Pushover - Get Pushover service settings](https://docs.gitlab.com/ee/api/services.html#get-pushover-service-settings)
- [ ] [Redmine - Create/Edit Redmine service](https://docs.gitlab.com/ee/api/services.html#create-edit-redmine-service)
- [ ] [Redmine - Delete Redmine service](https://docs.gitlab.com/ee/api/services.html#delete-redmine-service)
- [ ] [Redmine - Get Redmine service settings](https://docs.gitlab.com/ee/api/services.html#get-redmine-service-settings)
- [ ] [Slack notifications - Create/Edit Slack service](https://docs.gitlab.com/ee/api/services.html#create-edit-slack-service)
- [ ] [Slack notifications - Delete Slack service](https://docs.gitlab.com/ee/api/services.html#delete-slack-service)
- [ ] [Slack notifications - Get Slack service settings](https://docs.gitlab.com/ee/api/services.html#get-slack-service-settings)
- [ ] [Microsoft Teams - Create/Edit Microsoft Teams service](https://docs.gitlab.com/ee/api/services.html#create-edit-microsoft-teams-service)
- [ ] [Microsoft Teams - Delete Microsoft Teams service](https://docs.gitlab.com/ee/api/services.html#delete-microsoft-teams-service)
- [ ] [Microsoft Teams - Get Microsoft Teams service settings](https://docs.gitlab.com/ee/api/services.html#get-microsoft-teams-service-settings)
- [ ] [Mattermost notifications - Create/Edit Mattermost notifications service](https://docs.gitlab.com/ee/api/services.html#create-edit-mattermost-notifications-service)
- [ ] [Mattermost notifications - Delete Mattermost notifications service](https://docs.gitlab.com/ee/api/services.html#delete-mattermost-notifications-service)
- [ ] [Mattermost notifications - Get Mattermost notifications service settings](https://docs.gitlab.com/ee/api/services.html#get-mattermost-notifications-service-settings)
- [ ] [JetBrains TeamCity CI - Create/Edit JetBrains TeamCity CI service](https://docs.gitlab.com/ee/api/services.html#create-edit-jetbrains-teamcity-ci-service)
- [ ] [JetBrains TeamCity CI - Delete JetBrains TeamCity CI service](https://docs.gitlab.com/ee/api/services.html#delete-jetbrains-teamcity-ci-service)
- [ ] [JetBrains TeamCity CI - Get JetBrains TeamCity CI service settings](https://docs.gitlab.com/ee/api/services.html#get-jetbrains-teamcity-ci-service-settings)
- [ ] [Jenkins CI - Create/Edit Jenkins CI service](https://docs.gitlab.com/ee/api/services.html#create-edit-jenkins-ci-service)
- [ ] [Jenkins CI - Delete Jenkins CI service](https://docs.gitlab.com/ee/api/services.html#delete-jenkins-ci-service)
- [ ] [Jenkins CI - Get Jenkins CI service settings](https://docs.gitlab.com/ee/api/services.html#get-jenkins-ci-service-settings)
- [ ] [Jenkins CI (Deprecated) Service - Create/Edit Jenkins CI (Deprecated) service](https://docs.gitlab.com/ee/api/services.html#create-edit-jenkins-ci-deprecated-service)
- [ ] [Jenkins CI (Deprecated) Service - Delete Jenkins CI (Deprecated) service](https://docs.gitlab.com/ee/api/services.html#delete-jenkins-ci-deprecated-service)
- [ ] [Jenkins CI (Deprecated) Service - Get Jenkins CI (Deprecated) service settings](https://docs.gitlab.com/ee/api/services.html#get-jenkins-ci-deprecated-service-settings)
- [ ] [MockCI - Create/Edit MockCI service](https://docs.gitlab.com/ee/api/services.html#create-edit-mockci-service)
- [ ] [MockCI - Delete MockCI service](https://docs.gitlab.com/ee/api/services.html#delete-mockci-service)
- [ ] [MockCI - Get MockCI service settings](https://docs.gitlab.com/ee/api/services.html#get-mockci-service-settings)

# Settings
- [ ] [Get current application settings](https://docs.gitlab.com/ee/api/settings.html#get-current-application-settings)
- [ ] [Change application settings](https://docs.gitlab.com/ee/api/settings.html#change-application-settings)

# Sidekiq metrics
- [ ] [Get the current Queue Metrics](https://docs.gitlab.com/ee/api/sidekiq_metrics.html#get-the-current-queue-metrics)
- [ ] [Get the current Process Metrics](https://docs.gitlab.com/ee/api/sidekiq_metrics.html#get-the-current-process-metrics)
- [ ] [Get the current Job Statistics](https://docs.gitlab.com/ee/api/sidekiq_metrics.html#get-the-current-job-statistics)
- [ ] [Get a compound response of all the previously mentioned metrics](https://docs.gitlab.com/ee/api/sidekiq_metrics.html#get-a-compound-response-of-all-the-previously-mentioned-metrics)

# System Hooks
- [ ] [List system hooks](https://docs.gitlab.com/ee/api/system_hooks.html#list-system-hooks)
- [ ] [Add new system hook](https://docs.gitlab.com/ee/api/system_hooks.html#add-new-system-hook)
- [ ] [Test system hook](https://docs.gitlab.com/ee/api/system_hooks.html#test-system-hook)
- [ ] [Delete system hook](https://docs.gitlab.com/ee/api/system_hooks.html#delete-system-hook)

# Tags
- [ ] [List project repository tags](https://docs.gitlab.com/ee/api/tags.html#list-project-repository-tags)
- [ ] [Get a single repository tag](https://docs.gitlab.com/ee/api/tags.html#get-a-single-repository-tag)
- [ ] [Create a new tag](https://docs.gitlab.com/ee/api/tags.html#create-a-new-tag)
- [ ] [Delete a tag](https://docs.gitlab.com/ee/api/tags.html#delete-a-tag)
- [ ] [Create a new release](https://docs.gitlab.com/ee/api/tags.html#create-a-new-release)
- [ ] [Update a release](https://docs.gitlab.com/ee/api/tags.html#update-a-release)

# Todos
- [x] [Get a list of todos](https://docs.gitlab.com/ee/api/todos.html#get-a-list-of-todos)
- [x] [Mark a todo as done](https://docs.gitlab.com/ee/api/todos.html#mark-a-todo-as-done)
- [x] [Mark all todos as done](https://docs.gitlab.com/ee/api/todos.html#mark-all-todos-as-done)

# Users
- [x] [List users - For normal users](https://docs.gitlab.com/ee/api/users.html#for-normal-users)
- [x] [List users - For admins](https://docs.gitlab.com/ee/api/users.html#for-admins)
- [x] [Single user - For user](https://docs.gitlab.com/ee/api/users.html#for-user)
- [x] [Single user - For admin](https://docs.gitlab.com/ee/api/users.html#for-admin)
- [x] [User creation](https://docs.gitlab.com/ee/api/users.html#user-creation)
- [ ] [User modification](https://docs.gitlab.com/ee/api/users.html#user-modification)
- [ ] [User deletion](https://docs.gitlab.com/ee/api/users.html#user-deletion)
- [x] [User status](https://docs.gitlab.com/ee/api/users.html#user-status)
- [x] [Get the status of a user](https://docs.gitlab.com/ee/api/users.html#get-the-status-of-a-user)
- [x] [Set user status](https://docs.gitlab.com/ee/api/users.html#set-user-status)
- [x] [List SSH keys](https://docs.gitlab.com/ee/api/users.html#list-ssh-keys)
- [x] [List SSH keys for user](https://docs.gitlab.com/ee/api/users.html#list-ssh-keys-for-user)
- [x] [Single SSH key](https://docs.gitlab.com/ee/api/users.html#single-ssh-key)
- [x] [Add SSH key](https://docs.gitlab.com/ee/api/users.html#add-ssh-key)
- [x] [Add SSH key for user](https://docs.gitlab.com/ee/api/users.html#add-ssh-key-for-user)
- [x] [Delete SSH key for current user](https://docs.gitlab.com/ee/api/users.html#delete-ssh-key-for-current-user)
- [ ] [Delete SSH key for given user](https://docs.gitlab.com/ee/api/users.html#delete-ssh-key-for-given-user)
- [ ] [List all GPG keys](https://docs.gitlab.com/ee/api/users.html#list-all-gpg-keys)
- [ ] [Get a specific GPG key](https://docs.gitlab.com/ee/api/users.html#get-a-specific-gpg-key)
- [ ] [Add a GPG key](https://docs.gitlab.com/ee/api/users.html#add-a-gpg-key)
- [ ] [Delete a GPG key](https://docs.gitlab.com/ee/api/users.html#delete-a-gpg-key)
- [ ] [List all GPG keys for given user](https://docs.gitlab.com/ee/api/users.html#list-all-gpg-keys-for-given-user)
- [ ] [Get a specific GPG key for a given user](https://docs.gitlab.com/ee/api/users.html#get-a-specific-gpg-key-for-a-given-user)
- [ ] [Add a GPG key for a given user](https://docs.gitlab.com/ee/api/users.html#add-a-gpg-key-for-a-given-user)
- [ ] [Delete a GPG key for a given user](https://docs.gitlab.com/ee/api/users.html#delete-a-gpg-key-for-a-given-user)
- [ ] [List emails](https://docs.gitlab.com/ee/api/users.html#list-emails)
- [ ] [List emails for user](https://docs.gitlab.com/ee/api/users.html#list-emails-for-user)
- [ ] [Single email](https://docs.gitlab.com/ee/api/users.html#single-email)
- [ ] [Add email](https://docs.gitlab.com/ee/api/users.html#add-email)
- [ ] [Add email for user](https://docs.gitlab.com/ee/api/users.html#add-email-for-user)
- [ ] [Delete email for current user](https://docs.gitlab.com/ee/api/users.html#delete-email-for-current-user)
- [ ] [Delete email for given user](https://docs.gitlab.com/ee/api/users.html#delete-email-for-given-user)
- [ ] [Block user](https://docs.gitlab.com/ee/api/users.html#block-user)
- [ ] [Unblock user](https://docs.gitlab.com/ee/api/users.html#unblock-user)
- [ ] [Get all impersonation tokens of a user](https://docs.gitlab.com/ee/api/users.html#get-all-impersonation-tokens-of-a-user)
- [ ] [Get an impersonation token of a user](https://docs.gitlab.com/ee/api/users.html#get-an-impersonation-token-of-a-user)
- [x] [Create an impersonation token](https://docs.gitlab.com/ee/api/users.html#create-an-impersonation-token)
- [ ] [Revoke an impersonation token](https://docs.gitlab.com/ee/api/users.html#revoke-an-impersonation-token)
- [ ] [Revoke an impersonation token - Get user activities (admin only)](https://docs.gitlab.com/ee/api/users.html#get-user-activities-admin-only)

# Validate CI configuration
- [ ] [Validate the .gitlab-ci.yml (API)](https://docs.gitlab.com/ee/api/lint.html#validate-the-gitlab-ci-yml-api)

# Version
- [x] [Version API](https://docs.gitlab.com/ee/api/version.html#version-api)

# Wikis
- [x] [List wiki pages](https://docs.gitlab.com/ee/api/wikis.html#list-wiki-pages)
- [x] [Get a wiki page](https://docs.gitlab.com/ee/api/wikis.html#get-a-wiki-page)
- [x] [Create a new wiki page](https://docs.gitlab.com/ee/api/wikis.html#create-a-new-wiki-page)
- [x] [Edit an existing wiki page](https://docs.gitlab.com/ee/api/wikis.html#edit-an-existing-wiki-page)
- [x] [Delete a wiki page](https://docs.gitlab.com/ee/api/wikis.html#delete-a-wiki-page)
- [ ] [Upload an attachment to the wiki repository](https://docs.gitlab.com/ee/api/wikis.html#upload-an-attachment-to-the-wiki-repository)
